extensions/amp-ad/0.1/test/test-amp-ad-3p-impl.js

```diff
@@ -357,4 +357,171 @@ describes.realWin('amp-ad-3p-impl', {
       });
     });
   });
+
+  describe('full width responsive ad', () => {
+
+    const VIEWPORT_WIDTH = 300;
+    const VIEWPORT_HEIGHT = 600;
+
+    beforeEach(() => {
+      adConfig['_ping_'].fullWidthHeightRatio = 1.2;
+      win.document.body.removeChild(ad3p.element);
+    });
+
+    describe('should resize correctly', () => {
+      let iframe;
+      let impl;
+      let element;
+      let attemptChangeSizeSpy;
+
+      function constructImpl(config) {
+        iframe = win.document.createElement('iframe');
+
+        config.type = '_ping_';
+        element = createElementWithAttributes(win.document, 'amp-ad', config);
```

Comment:
seems like some code duplication with `createAmpAd` function at the top of this file.

Reply:
Ack, (also see https://github.com/ampproject/amphtml/pull/12200#discussion_r153683396). 
I don't really see how we can share much code here in a meaningful way but open to suggestions.

Reply:
for instance, you should not create an iframe and append it to the amp-ad element manually. and i'm not seeing you read this iframe anyway.

Reply:
Done.